src/map/itemdb.cpp

```diff
@@ -1932,142 +1917,164 @@ static bool itemdb_read_noequip(char* str[], int columns, int current) {
 	return true;
 }
 
-/**
- * @return: amount of retrieved entries.
- **/
-static int itemdb_combo_split_atoi (char *str, t_itemid *val) {
-	int i;
+bool ComboDatabase::parseComboNode(const YAML::Node &node, const std::string &nodeName, std::vector<std::vector<t_itemid>> &items_list) {
+	if (!this->nodesExist(node, { nodeName })) {
+		return 0;
+	}
 
-	for (i=0; i<MAX_ITEMS_PER_COMBO; i++) {
-		if (!str) break;
+	if (!node[nodeName].IsSequence()) {
+		this->invalidWarning(node[nodeName], "%s should be a sequence.\n", nodeName.c_str());
+		return 0;
+	}
 
-		val[i] = strtoul(str, nullptr, 10);
+	std::vector<t_itemid> items = {};
 
-		str = strchr(str,':');
+	for (const auto &it : node[nodeName]) {
+		std::string item_name = it.as<std::string>();
 
-		if (str)
-			*str++=0;
-	}
+		std::shared_ptr<item_data> item = item_db.search_aegisname(item_name.c_str());
 
-	if( i == 0 ) //No data found.
-		return 0;
-
-	return i;
-}
+		if (item == nullptr) {
+			this->invalidWarning(it, "Invalid item %s, skipping.\n", item_name.c_str());
+			return false;
+		}
 
-/**
- * <combo{:combo{:combo:{..}}}>,<{ script }>
- **/
-static void itemdb_read_combos(const char* basedir, bool silent) {
-	uint32 lines = 0, count = 0;
-	char line[1024];
+		items.push_back(item->nameid);
+	}
 
-	char path[256];
-	FILE* fp;
+	if (items.empty()) {
+		this->invalidWarning(node[nodeName], "Empty combo, skipping.\n");
+		return false;
+	}
 
-	sprintf(path, "%s/%s", basedir, "item_combo_db.txt");
+	if (items.size() < 2) {
+		this->invalidWarning(node[nodeName], "Not enough item to make a combo (need at least 2). Skipping.\n");
+		return false;
+	}
 
-	if ((fp = fopen(path, "r")) == NULL) {
-		if(silent==0) ShowError("itemdb_read_combos: File not found \"%s\".\n", path);
-		return;
+	if (items.size() > MAX_ITEMS_PER_COMBO) {
```

Comment:
Do we need this still? Since it’s a vector we shouldn’t be concerned of a max limit anymore.

Reply:
Yeah, I say remove it, if we can just go back and verify we don't have any other limiters, why cap it? :)